File: OCaml/0002.ml
(* Function to compute the sum of even Fibonacci numbers up to a given maximum value *)
let sum_even_fibonacci max_val =
  let rec aux term1 term2 sum_even =
    if term1 > max_val then
      sum_even
    else
      let sum_even = if term1 mod 2 = 0 then sum_even + term1 else sum_even in
      let next_term = term1 + term2 in
      aux term2 next_term sum_even
  in
  aux 1 2 0

(* Define the maximum value *)
let max_value = 4000000

(* Calculate and print the result *)
let () =
  let result = sum_even_fibonacci max_value in
  Printf.printf "Sum of the even-valued terms: %d\n" result

File: Julia/0002.jl
# Define the maximum value
const MAX_VALUE = 4000000

# Function to compute the sum of even Fibonacci numbers up to a given maximum value
function sum_even_fibonacci(max_val)
    term1 = 1
    term2 = 2
    sum_even = 0

    while term1 <= max_val
        # Check if the current term is even
        if term1 % 2 == 0
            sum_even += term1
        end

        # Generate the next term in the Fibonacci sequence
        next_term = term1 + term2
        term1 = term2
        term2 = next_term
    end

    return sum_even
end

# Compute the result
result = sum_even_fibonacci(MAX_VALUE)

# Output the result
println("Sum of the even-valued terms: ", result)
